### PR TITLE
Implement Mul

### PR DESCRIPTION
**Pull request by yacovm**
Signed-off-by: Yacov Manevich <yacovm@il.ibm.com>

### CODE REVIEW

File: driver/gurvy/bn254.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ func (z *bn254Zr) Plus(a driver.Zr) driver.Zr {
 	return &bn254Zr{new(big.Int).Add(z.Int, a.(*bn254Zr).Int)}
 }
 
+func (z *bn254Zr) Mul(a driver.Zr) driver.Zr {
+	prod := new(big.Int).Mul(z.Int, a.(*bn254Zr).Int)
+	return &bn254Zr{new(big.Int).Mod(prod, fr.Modulus())}
```
Comment (ale-linux) [inline driver/gurvy/bn254.go:36]: this new here is redundant isn't it?

File: math_test.go [hunk 2]
```diff
@@ -276,6 +282,18 @@ func runModAddSubNegTest(t *testing.T, c *Curve) {
 	assert.True(t, bagain.Equals(b))
 }
 
+func runMulTest(t *testing.T, c *Curve) {
+	rng, err := c.Rand()
+	assert.NoError(t, err)
+
+	r := c.NewRandomZr(rng)
```
Comment (ale-linux) [inline math_test.go:289]: can we please add a D/H style test as well - like that a^b^c = a^c^b = a^bc?

File: driver/gurvy/bn254.go (hunk 1 above)
Comment (yacovm) [reply driver/gurvy/bn254.go]: fixed

File: math_test.go (hunk 2 above)
Comment (yacovm) [reply math_test.go:289]: Let me know if it's OK

**Review by ale-linux — APPROVED** [review]

**Pull request merged**